client/src/main/fbthrift/com/facebook/thrift/transport/THttp2Client.java

```diff
@@ -0,0 +1,160 @@
+/* Copyright (c) 2023 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+package com.facebook.thrift.transport;
+
+import static java.time.LocalTime.now;
+import java.io.ByteArrayOutputStream;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Protocol;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.units.qual.A;
+
+public class THttp2Client extends TTransport {
```

Comment:
Is this class designed to be used in single-thread?

Reply:
Yes, it can not be used in multi-threads.